examples/api/lib/widgets/form/form.1.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+/// This sample demonstrates showing a confirmation dialog when the user
+/// attempts to navigate away from a page with unsaved [Form] data.
+
+void main() => runApp(_MyApp());
+
+class _MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: _HomePage(),
+    );
+  }
+}
+
+class _HomePage extends StatefulWidget {
+  @override
+  State<_HomePage> createState() => _HomePageState();
+}
+
+class _HomePageState extends State<_HomePage> {
+  final TextEditingController _controller = TextEditingController();
+  String _savedValue = '';
+
+  @override
+  void initState() {
+    super.initState();
+    _controller.addListener(_onChanged);
+  }
+
+  @override
+  void dispose() {
+    _controller.removeListener(_onChanged);
+    super.dispose();
+  }
+
+  void _onChanged() {
+    setState(() {});
+  }
+
+  Future<void> _showDialog() async {
+    final bool? shouldDiscard = await showDialog<bool>(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('Are you sure?'),
+          content: const Text('Any unsaved changes will be lost!'),
+          actions: <Widget>[
+            TextButton(
+              child: const Text('Yes, discard my changes'),
+              onPressed: () {
+                Navigator.pop(context, true);
+              },
+            ),
+            TextButton(
+              child: const Text('No, continue editing'),
+              onPressed: () {
+                Navigator.pop(context, false);
+              },
+            ),
+          ],
+        );
+      },
+    );
+
+    if (shouldDiscard ?? false) {
+      SystemNavigator.pop();
```

Comment:
That's right.  I'm exiting the app just because in this specific example, there are no other routes.  If I change this to `Navigator.of(context).pop();`, then it will pop to a black screen in this case.

Is there something else I could call that would be more universally useful here?

Reply:
Navigator.of(context).maybePop is the answer here, but we still need to call SystemNavigator

Reply:
As discussed elsewhere, I'm going to leave a comment here explaining the situation. 
 Ideally we'd have something like https://github.com/flutter/flutter/issues/11490 instead.